arch/risc-v/src/common/riscv_macros.S

```diff
@@ -362,6 +362,9 @@
   REGLOAD \out, RISCV_PERCPU_HARTID(\out)
 #else
   csrr    \out, CSR_MHARTID
+#  if defined(CONFIG_ARCH_RV_HARTID_BASE) && CONFIG_ARCH_RV_HARTID_BASE > 0
```

Comment:
why not remove ARCH_RV_GLOBAL_HARTID if we skip adjust when CONFIG_ARCH_RV_HARTID_BASE equals?